cw_infrastructure/cw_infrastructure/check_config_values.py

```diff
@@ -0,0 +1,536 @@
+    """Utility method to print error messages to stderr.
+
+       @param option_name - The name of the option the error relates to.
+       @param message     - A description of the problem"""
+    sys.stderr.write("ERROR: {}: {}\n".format(option_name, message))
+
+
+def warning(option_name, message):
+    """Utility method to print warning messages to stderr.
+
+       @param option_name - The name of the option the warning relates to.
+       @param message     - A description of the problem"""
+    sys.stderr.write("WARN: {}: {}\n".format(option_name, message))
+
+
+def ip_version(value):
+    """Return the IP version of the supplied IP address, which is passed as a
+    string. If the argument is not an IP address this function returns None,
+    which is more convenient than raising an exception"""
+    try:
+        socket.inet_pton(socket.AF_INET, value)
+        return 4
+    except socket.error:
+        try:
+            socket.inet_pton(socket.AF_INET6, value)
+            return 6
+        except socket.error:
+            return None
+
+
+def is_ip_addr(value):
+    """Return whether the supplied string is a valid IP address"""
+    return (ip_version(value) is not None)
+
+
+def is_domain_name(value):
+    """Return whether the supplied string is a valid domain name"""
+
+    # A domain consists of one or more labels separated by dots.  Each label
+    # must start and end with a letter or number. The characters in between can
+    # be either letters, numbers or a hyphen. In addition a label can be at most
+    # 63 characters long, and the address as a whole can be 255 characters long.
+    #
+    # Note that RFC 1035, section 2.3.1 technically forbids labels from starting
+    # with a digit. However this does happen in practice, so we allow it.
+    #
+    # The following code builds a regex that only matches on valid domain names.
+    # The only thing it does not police is the total length of the name, which
+    # is checked separately below.
+    label_regex = r"[a-zA-Z\d]([a-zA-Z\d-]{0,61}[a-zA-Z\d])?"
+    domain_regex = re.compile(r"^({0})(\.{0})*$".format(label_regex))
+
+    if len(value) > 255 or not domain_regex.match(value):
+        return False
+    else:
+        return True
+
+
+def is_resolvable_domain_name(value):
+    """Return whether the supplied string is a resolvable domain name"""
+    try:
+        socket.gethostbyname(value)
+        return True
+    except socket.gaierror:
+        return False
+
+
+def is_naptr_resolvable(naptr):
+    """Check whether an NAPTR domain has any records"""
+
+    return is_domain_resolvable(naptr, 'NAPTR')
+
+
+def is_srv_resolvable(srv):
+    """Check whether an SRV domain has any records"""
+
+    return is_domain_resolvable(srv, 'SRV')
+
+
+def is_domain_resolvable(name, rrtype):
+    """Check whether the given domain has any records of the given type"""
+
+    try:
+        answers = dns.resolver.query(name, rrtype)
+        return len(answers) != 0
+
+    except:
```

Comment:
These 'except' blocks may be too generic. Due to the missing dns.resolver import dns.resolver.query threw a NameError which was caught by the except block, causing the script to report a valid SRV record as invalid.